opendbc/can/common.h

```diff
@@ -28,6 +33,9 @@ unsigned int xor_checksum(uint32_t address, const Signal &sig, const std::vector
 unsigned int hkg_can_fd_checksum(uint32_t address, const Signal &sig, const std::vector<uint8_t> &d);
 unsigned int fca_giorgio_checksum(uint32_t address, const Signal &sig, const std::vector<uint8_t> &d);
 unsigned int pedal_checksum(uint32_t address, const Signal &sig, const std::vector<uint8_t> &d);
+unsigned int tesla_checksum(uint32_t address, const Signal &sig, const std::vector<uint8_t> &d);
+
+bool tesla_is_counter(const Signal &sig);
```

Comment:
@jyoung8607 does this kind of pattern enable any kind of refactors/unblock anything you've been wanting to do?

Reply:
Actually it would be better to have one post init/setup hook for each DBC that can set things and run checks:

https://github.com/commaai/opendbc/blob/295be61b97b5bf92c3a3b656530a12b1562a0523/opendbc/can/dbc.cc#L94-L101

What do you think>

Reply:
A long time ago I proposed [Vector compliant message attributes](https://github.com/commaai/opendbc/pull/633#discussion_r910897364) that could be *added* to a DBC, without changing any of the existing naming or other content.

That said, now that car ports are part of opendbc, just a post init/setup hook where we can encapsulate all the manufacturer specific stuff would be easier, and wouldn't require any DBC changes at all.

I think I like your solution better.